crates/ruff/src/args.rs

```diff
@@ -728,7 +729,7 @@ impl CheckCommand {
             preview: resolve_bool_arg(self.preview, self.no_preview).map(PreviewMode::from),
             respect_gitignore: resolve_bool_arg(self.respect_gitignore, self.no_respect_gitignore),
             select: self.select,
-            target_version: self.target_version,
+            target_version: self.target_version.map(AstPythonVersion::from),
```

Comment:
```suggestion
            target_version: self.target_version.into(),
```

Reply:
hehe, my nit is basically the same as @dhruvmanila's in https://github.com/astral-sh/ruff/pull/16170/files#r1959159116 😄